From e90e3d2a3e345385e2bfb95dd4d6ae30e5c0c9f9 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Fri, 21 Mar 2025 16:04:18 +0800
Subject: [PATCH 434/652] drm: eswin: es_dw_hdmi: allow non-standard pixclk

The mode_valid function currently only allows pixel clocks that exactly
matches existing configurations.

Switch to use <= instead of == to allow non-standard pixel clocks (but
overclocking is still disallowed).

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 drivers/gpu/drm/eswin/es_dw_hdmi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/eswin/es_dw_hdmi.c b/drivers/gpu/drm/eswin/es_dw_hdmi.c
index eefbabba3c92b..ae389aca75eb2 100644
--- a/drivers/gpu/drm/eswin/es_dw_hdmi.c
+++ b/drivers/gpu/drm/eswin/es_dw_hdmi.c
@@ -260,7 +260,7 @@ dw_hdmi_eswin_mode_valid(struct dw_hdmi *hdmi, void *data,
 	int i;
 
 	for (i = 0; mpll_cfg[i].mpixelclock != (~0UL); i++) {
-		if (pclk == mpll_cfg[i].mpixelclock) {
+		if (pclk <= mpll_cfg[i].mpixelclock) {
 			valid = true;
 			break;
 		}
-- 
2.53.0

